packages/compute/src/vm.js

```diff
@@ -809,6 +829,139 @@ VM.prototype.request = function(reqOpts, callback) {
   });
 };
 
+/**
+ * Poll `getMetadata` to check the VM's status. This runs a loop to ping
+ * the API on an interval.
+ *
+ * Note: This method is automatically called when a `waitFor()` call
+ * is made.
+ *
+ * @private
+ */
+VM.prototype.startPolling_ = function() {
+  var self = this;
+
+  if (!this.hasActiveWaiters) {
+    return;
+  }
+
+  this.getMetadata(function(err, metadata) {
+    var now = new Date();
+    for (var index in self.waiters) {
+      if (err) {
+        self.waiters[index].callback(err, null);
+        self.waiters.splice(index, 1);
+      } else if (metadata && metadata.status === self.waiters[index].status) {
+        self.waiters[index].callback(null, metadata);
+        self.waiters.splice(index, 1);
+      } else {
+        if ((now.getTime() / 1000) - self.waiters[index].startTime >=
+              self.waiters[index].timeout) {
+          self.waiters[index].callback({
+            name: 'WaitForTimeout'
+          },null);
+          self.waiters.splice(index, 1);
+        }
+      }
+    }
+    if (self.waiters.length <= 0) {
+      self.hasActiveWaiters = false;
+    } else {
+      setTimeout(self.startPolling_.bind(self), WAIT_FOR_POLLING_INTERVAL);
+    }
+  });
+};
+
+/**
+ * This function will callback when the VM is in the specified state.
+ *
+ * Will time out after the specified time (default: 300 seconds).
+ *
+ * @param {string} status - The status to wait for. This can be:
+ *  - "PROVISIONING"
+ *  - "STAGING"
+ *  - "RUNNING"
+ *  - "STOPPING"
+ *  - "SUSPENDING"
+ *  - "SUSPENDED"
+ *  - "TERMINATED"
+ * @param {object=} options - Configuration object.
+ * @param {number} options.timeout - The number of seconds to wait until
```

Comment:
Wait, sorry, this won't work. I only have one polling interval running at any one time to minimize the number of API requests, and it checks every waiter on each API response. If we add this, we'd have to either :
a) change the polling interval for every `waitFor()` currently waiting after each `waitFor()` call with an `options.timeout`, which would confuse users, or 
b) change it so it polls for each individual `waitFor()` call, which would increase the amount of superfluous API requests.
Neither of these is desirable. Besides, we probably don't want to go any lower than 2 seconds for this, and user will probably not want to set the interval for anything longer than they have to.